### PR TITLE
fix: ensure safe math in executor and sdk

### PR DESCRIPTION
**Pull request by palozano**
- feat: update risk reward params in side effect
- refactor: from number to bn
- refactor: finish refactor to bn
- refactor: number in sdk to bn
- refactor: safe conversion amountconverter
- chore: add tests and fix func
- chore: run formating
- ci: add sdk tests

# Summary of changes

Addresses issue #574:

- Updates the Risk/Reward parameters in SideEffect to use BN.js.
- Uses BN.js operations with large numbers.
- Adds a check for not overflowing conversion operations.
- Adds tests for AmountConverter in SDK.

If it fixes a bug or creates a feature, link the issue.

## Acceptance Checklist

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue):
  - [X] Have you **written tests**to protect against future occurrences of the bug?
- [ ] New feature (non-breaking change which adds functionality)
  - [ ] Have you **seen it working** in a development environment?
  - [X] Have you **written tests** for **happy** and **unhappy** paths?
  - [ ] Have you implemented this feature as per the **issue acceptance criteria**?
- [ ] Breaking change (a feature that would cause existing functionality not to work as expected
  - [ ] Is the breaking change **documented**?
  - [ ] Has any previous changes been **deprecated**?
- [ ] CI/CD
  - [ ] If introducing new actions, have you **looked at the action code** for anything spurious?
  - [ ] Have you written **custom scripts** for things that could be actions already on the marketplace?
  - [ ] If introducing new actions, have you **pegged a static version**?
- [ ] Documentation Update
  - [ ] Have you checked other documentation, such as the docs repository?
  - [ ] If updating script documentation, have you **tested it on both environments**?
- [ ] Substrate
  - [ ] RPC methods?
  - [ ] Chainspec, both JSON specs & the module?
  - [ ] Runtime versioning?
  - [ ] Benchmarks?
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Tests can, and should, be extended.

### CODE REVIEW

File: client/packages/executor/src/executionManager/sideEffect.ts [hunk 1]
```diff
@@ -173,8 +175,8 @@ export class SideEffect extends EventEmitter {
             this.target = new TextDecoder().decode(sideEffect.target.toU8a())
             this.gateway = sdk.gateways[this.target]
             this.securityLevel = this.evalSecurityLevel(this.gateway.gatewayType)
-            this.reward = new BehaviorSubject(sdk.circuit.toFloat(sideEffect.maxReward)) // this is always in TRN (native asset)
-            this.insurance = sdk.circuit.toFloat(sideEffect.insurance) // this is always in TRN (native asset)
+            this.reward = new BehaviorSubject(new BN.BN(sdk.circuit.toFloat(sideEffect.maxReward))) // this is always in TRN (native asset)
```
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:178]: It doesnt make a lot of sense to convert to float, and then to BN, we essentially loose all precision. BN supports no floats.

**Review by MaciejBaj — APPROVED** [review]

File: client/packages/executor/src/executionManager/sideEffect.ts [hunk 2]
```diff
@@ -273,13 +276,16 @@ export class SideEffect extends EventEmitter {
      * used to determine if another bid should be placed.
      */
     recomputeMaxProfit() {
-        const txCostUsd = this.gateway.toFloat(this.txCostNative.getValue()) * this.nativeAssetPrice.getValue()
+        const txCostUsd = Big(this.gateway.toFloat(this.txCostNative.getValue())).times(this.nativeAssetPrice.getValue())
```
Comment (petscheit) [inline client/packages/executor/src/executionManager/sideEffect.ts:279]: We run into the same problem discussed before here. We can't use `toFloat()` as it breaks precision.

Comment (petscheit) [reply client/packages/executor/src/executionManager/sideEffect.ts:279]: AmountConverter also doesn't contain a safe version of this functionality, which was outlined in the ticket.

File: client/packages/sdk/src/converters/amounts.ts [hunk 3]
```diff
@@ -94,33 +90,71 @@ export class AmountConverter {
...
+    }
+  }
+
+  /**
+   * Converts a price from integer (`BN` value) to USD (`number` value)
+   * 
+   * @param price The amount to be converted, with the correct amount of decimals
+   * @returns The price in USD
+   */
+  computeUsdPrice(price: Price): number {
```
Comment (petscheit) [inline client/packages/sdk/src/converters/amounts.ts:140]: This is good, and enables safe conversions

**Pull request closed without merging**